esapi/esapi_util/esys_iutil.c

```diff
@@ -0,0 +1,1202 @@
+                cp_hash_tab[*cpHashNum].alg =
+                    session->rsrc.misc.rsrc_session.authHash;
+                *cpHashNum += 1;
+            }
+        }
+    }
+    return r;
+}
+
+TSS2_RC
+iesys_compute_rp_hashtab(ESYS_CONTEXT * esys_context,
+                         TSS2L_SYS_AUTH_RESPONSE * rspAuths,
+                         const uint8_t * rpBuffer,
+                         size_t rpBuffer_size,
+                         HASH_TAB_ITEM rp_hash_tab[3], uint8_t * rpHashNum)
+{
+    uint8_t rcBuffer[4] = { 0 };
+    uint8_t ccBuffer[4];
+    TSS2_RC r = Tss2_Sys_GetCommandCode(esys_context->sys, &ccBuffer[0]);
+    return_if_error(r, "Error: get command code");
+
+    for (int i = 0; i < rspAuths->count; i++) {
+        RSRC_NODE_T *session = esys_context->session_tab[i];
+        bool rpHashFound = false;
+        for (int j = 0; j < *rpHashNum; j++)
+            if (rp_hash_tab[j].alg == session->rsrc.misc.rsrc_session.authHash) {
+                rpHashFound = true;
+                break;
+            }
+        if (!rpHashFound) {
+            rp_hash_tab[*rpHashNum].size = sizeof(TPMU_HA);
+            r = iesys_crypto_rpHash(session->rsrc.misc.rsrc_session.authHash,
+                                    rcBuffer, ccBuffer, rpBuffer, rpBuffer_size,
+                                    &rp_hash_tab[*rpHashNum].digest[0],
+                                    &rp_hash_tab[*rpHashNum].size);
+            if (r != TPM2_RC_SUCCESS)
+                return r;
+            rp_hash_tab[*rpHashNum].alg =
+                session->rsrc.misc.rsrc_session.authHash;
+            *rpHashNum += 1;
+        }
+    }
+    return TPM2_RC_SUCCESS;
+}
+
+TSS2_RC
+GetTpmHandle(ESYS_CONTEXT * esys_context,
+             ESYS_TR esys_handle, TPM2_RH * tpm_handle)
+{
+    RSRC_NODE_T *esys_object;
+    TSS2_RC r = esys_GetResourceObject(esys_context, esys_handle, &esys_object);
+    if (r != TPM2_RC_SUCCESS)
+        return r;
+    if (esys_object == NULL) {
+        LOG_ERROR("Handle for bad TR requested.");
+        return TSS2_ESYS_RC_BAD_TR;
+    }
+    *tpm_handle = esys_object->rsrc.handle;
+    return TPM2_RC_SUCCESS;
+}
+
+TSS2_RC
+esys_CreateResourceObject(ESYS_CONTEXT * esys_context,
+                          ESYS_TR esys_handle, RSRC_NODE_T ** esys_object)
+{
+    RSRC_NODE_T *new_esys_object = calloc(1, sizeof(RSRC_NODE_T));
+    if (new_esys_object == NULL)
+        return TSS2_ESYS_RC_MEMORY;
+    if (esys_context->rsrc_list == NULL) {
+        esys_context->rsrc_list = new_esys_object;
+        new_esys_object->next = NULL;
+    } else {
+        new_esys_object->next = esys_context->rsrc_list;
+        esys_context->rsrc_list = new_esys_object;
+    }
+    *esys_object = new_esys_object;
+    new_esys_object->esys_handle = esys_handle;
+    return TSS2_RC_SUCCESS;
+}
+
+TSS2_RC
+iesys_handle_to_tpm_handle(ESYS_TR esys_handle, TPM2_HANDLE * tpm_handle)
+{
+    if (esys_handle >= ESYS_TR_PCR0 && esys_handle <= ESYS_TR_PCR31) {
```

Comment:
checking to be sure `esys_handle` is `>=` to `ESYS_TR_PCR0` is checking to be sure an unsigned int is larger than `0` -> always true. This first part of the conditional can be omitted I think.